backend/yugioh/serializers.py

```diff
@@ -0,0 +1,101 @@
+from drf_spectacular.utils import extend_schema_field
+from rest_framework import serializers
+from .models import YugiohCard, YugiohCardInSet, YugiohCardSet, YugiohCardRarity
+
+
+class YugiohCardRaritySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = YugiohCardRarity
+        fields = ['rarity', 'rarity_code']
+
+
+class YugiohCardSetSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = YugiohCardSet
+        fields = ['card_set_name', 'set_code']
+        read_only_fields = ['card_set_name', 'set_code']
+
+
+class YugiohSerializer(serializers.ModelSerializer):
+    rarity = serializers.SerializerMethodField()
+    set = serializers.SerializerMethodField()
+
+    class Meta:
+        model = YugiohCard
+        fields = [
+            'id',
+            'card_name',
+            'rarity',
+            'set',
+            'type',
+            'frame_type',
+            'description',
+            'attack',
+            'defense',
+            'level',
+            'race',
+            'attribute',
+            'archetype',
+            'image',
+        ]
+        read_only_fields = ['id', 'set', 'rarity']
+        ordering_fields = ['id']
+
+    @staticmethod
+    @extend_schema_field(YugiohCardSetSerializer)
+    def get_rarity(obj):
+        cards = YugiohCardInSet.objects.filter(yugioh_card=obj)
+        all_rarities = YugiohCardInSetSerializer(cards, many=True).data
+        if len(all_rarities) == 0:
+            return None
+
+        rarity_ids = [rarity['rarity'] for rarity in all_rarities]
+        rarity_names = [
+            f'{YugiohCardRarity.objects.get(id=rarity_id).rarity}{YugiohCardRarity.objects.get(id=rarity_id).rarity_code}'
+            for rarity_id in rarity_ids]
+        return rarity_names
+
+    @staticmethod
+    @extend_schema_field(YugiohCardSetSerializer)
+    def get_set(obj):
+        sets = YugiohCardInSet.objects.filter(yugioh_card=obj)
+        all_sets = YugiohCardInSetSerializer(sets, many=True).data
+        if len(all_sets) == 0:
+            return None
+
+        all_sets_ids = [set_name['set'] for set_name in all_sets]
+        set_names = [
+            f'{YugiohCardSet.objects.get(id=set_id).card_set_name}({YugiohCardSet.objects.get(id=set_id).set_code})' for
+            set_id in all_sets_ids]
+        return set_names
+
+
+class YugiohCardInSetCardSerializer(YugiohSerializer):
+
+    class Meta(YugiohSerializer.Meta):
+        fields = [
+            'id',
+            'card_name',
+            'type',
+            'frame_type',
+            'description',
+            'attack',
+            'defense',
+            'level',
+            'race',
+            'attribute',
+            'archetype',
+            'image',
+        ]
+
+
+class YugiohCardInSetSerializer(serializers.ModelSerializer):
```

Comment:
Include the id of the rarity and the id of the set in the response.

Reply:
IDs of the rarity and set are included now.
The response looks like this one:

```
{
  "id": 50,
  "yugioh_card": {
    "id": 8,
    "card_name": "Cosmic Cyclone",
    "type": "Spell Card",
    "frame_type": "spell",
    "description": "Pay 1000 LP, then target 1 Spell/Trap on the field; banish it.",
    "attack": "NULL",
    "defense": "NULL",
    "level": "NULL",
    "race": "Quick-Play",
    "attribute": "NULL",
    "archetype": "NULL",
    "image": "https://images.ygoprodeck.com/images/cards/8267140.jpg"
  },
  "set": {
    "id": 6,
    "card_set_name": "Duel Devastator",
    "set_code": "DUDE"
  },
  "rarity": {
    "id": 4,
    "rarity": "Ultra Rare",
    "rarity_code": "(UR)"
  }
}
```